helmfile/charts/prometheus-alerts/templates/alerts/kubernetes-apps.yaml

```diff
@@ -53,7 +53,7 @@ spec:
       annotations:
         description: Container {{`{{`}} $labels.container {{`}}`}} in pod {{`{{`}} $labels.namespace {{`}}`}}/{{`{{`}} $labels.pod {{`}}`}} from {{`{{`}} $labels.cluster {{`}}`}} has been OOMKilled {{`{{`}} $value {{`}}`}} times in the last 10 minutes.
         summary: Kubernetes container OOMKilled.
-      expr: (kube_pod_container_status_restarts_total - kube_pod_container_status_restarts_total offset 10m >= 1) and ignoring (reason) min_over_time(kube_pod_container_status_last_terminated_reason{reason="OOMKilled"}[10m]) == 1
+      expr: (kube_pod_container_status_restarts_total - kube_pod_container_status_restarts_total offset 10m >= 2) and ignoring (reason) min_over_time(kube_pod_container_status_last_terminated_reason{reason="OOMKilled"}[10m]) == 2
```

Comment:
Have you tested this?

Reply:
If I understand the expression correctly, the first part counts the number of restarts and the second part checks that the last reason was OOMkilled.

Reply:
Thanks, I tested as a expression in prometheus. First expression checking if any container in pod restarted for atleast once within the last 10 minutes with greater than or equal to 2 and second expression checking if within the last 10 minutes, there was atleast two container in a pod terminated with the reason `OOMKilled` .

But yes i think we will have one container only also , so this should be 1.

Reply:
@OlleLarsson thanks !! It worked now and updated the metrics

![image](https://github.com/elastisys/compliantkubernetes-apps/assets/75951259/1df725ca-fee1-4895-afa8-d122bbd28a2d)

Reply:
I'm guessing that the alert will now trigger if a pod first restarts due to let's say Error and then restarts due to OOMKilled. Is it possible to make it so that it only triggers if all of the restarts was due to OOMKilled within the given timeframe?

Reply:
thanks !! let me check this possibility

Reply:
I think we can use this expression to give only restart with reason `OOMKilled` output 

**Includes all** 

```
(min_over_time(kube_pod_container_status_last_terminated_reason{reason!="OOMKilled"}[30m]) == 1) and on(container, pod) (kube_pod_container_status_restarts_total - kube_pod_container_status_restarts_total offset 30m >= 1)
```
![image](https://github.com/elastisys/compliantkubernetes-apps/assets/75951259/8cec01d9-592b-4641-b4fa-0e67cd46632a)

**Only OOMKilled**
```
(min_over_time(kube_pod_container_status_last_terminated_reason{reason="OOMKilled"}[30m]) == 1) and on(container, pod) (kube_pod_container_status_restarts_total - kube_pod_container_status_restarts_total offset 30m >= 1)
```
![image](https://github.com/elastisys/compliantkubernetes-apps/assets/75951259/9b29ad5c-3c11-4607-800f-7d52d2d4d47a)

Reply:
Isn't the second query functionally the same as the one you already had in the PR? Except that you here only check for one restart instead of two (I still want you to check for two).

If I read this correctly then it does not matter if you take the second one you list here or the one currently in the PR (if you change the number of restarts you check for to two). So then I vote for keeping the current query, just to make it easier to see what changed in this PR/commit.

Reply:
Sure we can keep the current query .